pkg/asset/templates/kubelet.yaml

```diff
@@ -27,17 +27,11 @@ spec:
         - --api-servers={{ index .APIServers 0 }}
         - --config=/etc/kubernetes/manifests
         - --allow-privileged
-        - --hostname-override=$(MY_POD_IP)
```

Comment:
You can't drop it either for the time being... see https://github.com/kubernetes/kubernetes/pull/29907 that fixes the problem

In newer releases, you may omit `HostnameOverride` I think. kubelet will use `os.Hostname()` as name and the IP of the default gateway as the external ip